addons/popochiu/engine/objects/room/popochiu_room.gd

```diff
@@ -442,11 +443,15 @@ func _move_along_path(distance: float, moving_character_data: Dictionary):
 				moving_character_data.character.update_scale()
 			return
 
-		distance -= distance_between_points
-		last_point = moving_character_data.path[0]
+		# We're at the next navigation point
+		distance_to_move -= distance_to_next_navigation_point
+		last_character_position = moving_character_data.path[0]
+		if moving_character_data.path.size() > 1:
+			moving_character_data.character.face_direction(moving_character_data.path[1])
```

Comment:
Not sure what's the expected outcome, can you please explain?

Reply:
moving_character_data.path[] will have a set of coordinates where the character is to reach. e.g.
0,0 then 100,200, then 50,300, then 400, 350.
When they're at 100,200 for example, [100,200] will be path[0]. The next direction they will walk is towards path[1] i.e. [50,300], so set their facing direction towards the next leg of the path.

Reply:
Imagine! I thought this was already the case... :facepalm: 
Thanks